app/registry_proto_conversions.go

```diff
@@ -0,0 +1,484 @@
+	case PackageTypeModule:
+		return packages.PackageType_PACKAGE_TYPE_MODULE
+	case PackageTypeSLAMMap:
+		return packages.PackageType_PACKAGE_TYPE_SLAM_MAP
+	case PackageTypeMLTraining:
+		return packages.PackageType_PACKAGE_TYPE_ML_TRAINING
+	}
+	return packages.PackageType_PACKAGE_TYPE_UNSPECIFIED
+}
+
+// Visibility specifies the type of visibility of a registry item.
+type Visibility int32
+
+const (
+	// VisibilityUnspecified represents an unspecified visibility.
+	VisibilityUnspecified Visibility = iota
+	// VisibilityPrivate are for registry items visible only within the owning org.
+	VisibilityPrivate
+	// VisibilityPublic are for registry items that are visible to everyone.
+	VisibilityPublic
+	// VisibilityPublicUnlisted are for registry items usable in everyone's robot but are hidden from the registry page as if they are private.
+	VisibilityPublicUnlisted
+)
+
+func visibilityFromProto(visibility pb.Visibility) Visibility {
+	switch visibility {
+	case pb.Visibility_VISIBILITY_UNSPECIFIED:
+		return VisibilityUnspecified
+	case pb.Visibility_VISIBILITY_PRIVATE:
+		return VisibilityPrivate
+	case pb.Visibility_VISIBILITY_PUBLIC:
+		return VisibilityPublic
+	case pb.Visibility_VISIBILITY_PUBLIC_UNLISTED:
+		return VisibilityPublicUnlisted
+	}
+	return VisibilityUnspecified
+}
+
+func visibilityToProto(visibility Visibility) pb.Visibility {
+	switch visibility {
+	case VisibilityUnspecified:
+		return pb.Visibility_VISIBILITY_UNSPECIFIED
+	case VisibilityPrivate:
+		return pb.Visibility_VISIBILITY_PRIVATE
+	case VisibilityPublic:
+		return pb.Visibility_VISIBILITY_PUBLIC
+	case VisibilityPublicUnlisted:
+		return pb.Visibility_VISIBILITY_PUBLIC_UNLISTED
+	}
+	return pb.Visibility_VISIBILITY_UNSPECIFIED
+}
+
+type isRegistryItemMetadata interface {
+	isRegistryItemMetadata()
+}
+
+type registryItemModuleMetadata struct {
+	ModuleMetadata *ModuleMetadata
+}
+
+type registryItemMLModelMetadata struct {
+	MlModelMetadata *MLModelMetadata
+}
+
+type registryItemMLTrainingMetadata struct {
+	MlTrainingMetadata *MLTrainingMetadata
+}
+
+func (*registryItemModuleMetadata) isRegistryItemMetadata() {}
+
+func (*registryItemMLModelMetadata) isRegistryItemMetadata() {}
+
+func (*registryItemMLTrainingMetadata) isRegistryItemMetadata() {}
+
+// ModuleMetadata holds the metadata of a module.
+type ModuleMetadata struct {
+	Models     []*Model
+	Versions   []*ModuleVersion
+	Entrypoint string
+	FirstRun   *string
+}
+
+func moduleMetadataFromProto(md *pb.ModuleMetadata) *ModuleMetadata {
+	var models []*Model
+	for _, version := range md.Models {
+		models = append(models, modelFromProto(version))
+	}
+	var versions []*ModuleVersion
+	for _, version := range md.Versions {
+		versions = append(versions, moduleVersionFromProto(version))
+	}
+	return &ModuleMetadata{
+		Models:     models,
+		Versions:   versions,
+		Entrypoint: md.Entrypoint,
+		FirstRun:   md.FirstRun,
+	}
+}
+
+// Model has the API and model of a model.
```

Comment:
(minor) this comment is a little confusing/recursive. What is the model of a model?

Reply:
Probably want messaging like this https://github.com/viamrobotics/api/blob/main/proto/viam/app/v1/app.proto#L1237-L1240.